Added default value field to toNumber function

kapilkumar-github opened the pull request:
Modified toNumber method to have two parameters
1) value
2) defaultValue - when given value can't be converted into number it will return the default value

falsyvalues committed:
Say goodbye to jscs.

jdalton committed:
Remove `countHolders`.

jdalton committed:
Use more destructuring.

jdalton committed:
Remove `baseNth`.

jdalton committed:
Remove coercion method use.

falsyvalues committed:
Fixin some leftovers bb7c959.

falsyvalues committed:
Prefere ES6 templates.

jdalton committed:
Remove `toString` coercion method use.

jdalton committed:
Remove `toUpper` and `toLower`.

jdalton committed:
Remove `toString` from `createCompounder`.

jdalton committed:
Remove `arrayReduce` from `createCompounder`.

falsyvalues committed:
Adjust styling of array checks in `head`, `uniq`, `uniqBy` and `uniqWith` to other methods.

jdalton committed:
Remove `toString` use from `createCaseFirst`.

jdalton committed:
Remove `toString` use from `castPath`.

jdalton committed:
Use more `Array#map`.

jdalton committed:
Remove `nativeKeys` and `nativeKeysIn`.

falsyvalues committed:
Fixin `keysIn` import in `getAllKeysIn` by switching to `baseKeysIn`.

jdalton committed:
Remove `getRawTag` and `objectToString`.

jdalton committed:
Simplify `flip`.

jdalton committed:
Simplify `tail`.

jdalton committed:
Remove `createCompounder`.

falsyvalues committed:
Implement `baseEach` and `baseEachRight` since `createBaseFor` is gone.

jdalton committed:
Simplify `words`.

jdalton committed:
Remove unused internal `createCtor`.

102 committed:
Remove stubTrue from cond example (#3085)

102 committed:
Remove constant function from examples (#3086)

merriam committed:
Update _.pad* examples to show pad length < string. (#3088)

One example now shows that specifying a padding length of less than the
length of the string returns the entire original string.

jdalton committed:
Reimplement find.

jdalton committed:
Add `someObj`.

falsyvalues committed:
Add `semi` option to eslint config.

falsyvalues committed:
ESify `find` implementation.

falsyvalues committed:
Remove semicolons.

jdalton committed:
Rename `someObj` to `someValues`.

jdalton committed:
Use more for-of

jdalton committed:
Remove `replaceHolders`.

falsyvalues committed:
Fixin declaration in `baseSum` (3e2b0bb).

falsyvalues committed:
Fixin declaration in `someValues` (571e752).

falsyvalues committed:
Fixin missing references in `repeat`.

falsyvalues committed:
Fixin scope range for var declarations in `hasPath` and `hasPathIn`.

falsyvalues committed:
Catch undeclared vars with eslint.

falsyvalues committed:
Math them all.

falsyvalues committed:
Use more for-of

falsyvalues committed:
Fixin scope range for var declarations in internal version of `hasPath`.

jdalton committed:
Remove internal `hasPath`.

jdalton committed:
Remove `toSource`.

jdalton committed:
Adjust spacing in template string expressions.

jdalton committed:
Split out `every` into `everyValue`.

jdalton committed:
Rename `someValues` to `someValue`.

falsyvalues committed:
Adjust eslint settings to new requirements (d10b44b).

falsyvalues committed:
Last spacing in template string expressions.

captbaritone committed:
Enable no-unexpected-multiline (#3103)

This rule can help catch errors introduced by automatic semicolon insertion
edge cases. It seems like a sensible thing to have enabled in a code base that
does not use semicolons.

http://eslint.org/docs/rules/no-unexpected-multiline

falsyvalues committed:
Proper `style` npm script execution (#3103).

jdalton committed:
Rename `mapKeys` and `mapValues` to `mapKey` and `mapValue`.

jdalton committed:
Fix jsdoc for `every` and `some`.

jdalton committed:
Split `map` out.

jdalton committed:
Simplify `mapKey` and `mapValue`.

jdalton committed:
Split `filter` out.

jdalton committed:
Fix typos.

jdalton committed:
Fix object coercion.

jdalton committed:
Remove `baseKeys` and `baseKeysIn`.

jdalton committed:
Fix typo in `find`.

jdalton committed:
Remove `baseSlice`.

falsyvalues committed:
Fixin new `map` imports.

falsyvalues committed:
Fixin `some` and `every` imports.

jdalton committed:
Remove `mergeData`.

jdalton committed:
Use `Array#splice` directly.

falsyvalues committed:
Fixin `filter` imports.

falsyvalues committed:
Fixin `repeat` import.

jdalton committed:
Update object checks.

greenberga committed:
Fix cloneDeep with circularly dependent Sets/Maps (#3123)

jdalton committed:
Cleanup baseClone.

prtksxna committed:
Fix CoC link in REAMDE (#3138)

falsyvalues committed:
Complete wrapper modules removal for `setToString` (c6854fa).

hargasinski committed:
Fix CoC link in .github/CONTRIBUTING.md (#3156)

falsyvalues committed:
Always use parentheses in arrow functions (fixin 67a3fb9).

falsyvalues committed:
Fixin tag scope in `baseClone`.

falsyvalues committed:
A better fix for `tag` scope in `baseClone`.

jdalton committed:
Support symbols in property paths of `set`. [closes #3189]

jisaacks committed:
default prefix to empty string (#3214)

This starting returning NaN for me. It looks like it is trying to add undefined + number when there is no prefix.

aretecode committed:
stringToPath: avoid shadowing variable (#3226)

- remove the shadowing done using `string` variable declared in upper scope

MingYinLv committed:
Update dropRight.js (#3241)

jdalton committed:
Avoid using the values toString method in _.invert if it’s not a function. [closes #3260]

falsyvalues committed:
Fixin loop args order regression from 3c2795b in `invert`.

falsyvalues committed:
Fixin loop args order regression from 3c2795b in `invertBy`.

nullvariable committed:
_.update() docs grammar fix (#3264)

aretecode committed:
avoid shadowing result (#3280)

related https://github.com/lodash/lodash/commit/e908519fc321829d6e43a1a40a791ce62e2fbc8c

aretecode committed:
Remove useless return in setToPairs. (#3286)

jdalton committed:
Fix typo in findKey.

Commit:
clarify documentation about sortedUniq (#3310)

clarify that sortedUniq only works on sorted arrays

falsyvalues committed:
Faster testing against first char in `stringToPath` (to close #3308).

rssteffey committed:
Fixes param definition (#3368)

Bitmask is number

chriscalo committed:
Add pending() function to debounce and throttle to fix #3387 (#3388)

jdalton committed:
Commit and whitespace nits in debounce.

Commit:
Verbiage change (#3408)

'T' missing in This.

drabinowitz committed:
Add missing keysIn import for baseClone.js (#3413)

LukasDrgon committed:
Add jsDelivr hits badge (#3418)

falsyvalues committed:
Remove semicolons from `dropRight`.

falsyvalues committed:
Remove trailing space in `sortedUniq`.

younesfkihi committed:
Fix typo in `parseInt` (#3433)

matthewdordal committed:
Ignore log files that end with random numbers (#3445)

younesfkihi committed:
Fix typo in `escapeRegExp` (#3448)

yuxino committed:
Fix #3466 (#3467)

yuxino committed:
Modify the logic to make the code more intuitive (#3472)

younesfkihi committed:
Fix typo in `inRange` (#3478)

nazar-pc committed:
Tiny performance improvement by not compiling regular expression each time (#3479)

timdorr committed:
Faster/simpler isPlainObject check (#3483)

Faster/simpler isPlainObject check.

This runs up the prototype chain to check equivalency. When run in a cross-realm environment (differing contexts, iframes, etc), this ensures we're checking the value's prototype matches its context-specific instance of Object. 

This is faster than calling `toString()` on the constructor. There's still the `baseGetTag()` call, which does its own `toString()`, but this swaps out one for cross-realm stuff. It's also a bit simpler to understand, I think.

falsyvalues committed:
Revert "Fix typo in `escapeRegExp` (#3448)"

This reverts commit 102d9e34d71838c027b5fc3d9153187f50842c95.

falsyvalues committed:
Tiny style correction in `createMathOperation`.

younesfkihi committed:
Remove `map` examples from `parseInt` and `trim` (#3487)

kybetter committed:
Fix operator in `createMathOperation`. (#3499)

GCHQDeveloper500 committed:
Update license file with MIT title (#3498)

TheLarkInn committed:
feat(sideEffects): Set sideEffects false for opt. (#3533)

This adds the webpack v4 sideEffects optimization. Since we are now in alpha I wanted to help ensure this gets out there so people testing webpack v4 can see the instant wins. <3

AlexanderLukin committed:
Remove duplicated "by which" words from the doc (#3534)

jdalton committed:
Enable _.words to detect ordinals in compound words. [closes #3561]

mattlubner committed:
rAF as default for debounce/throtte (#3560) (#3567)

Uses rAF by default for debounce/throtte unless `wait` is specified. (#3560)

jdalton committed:
Update argument count for `cloneWith` customizer. [closes #3582]

jdalton committed:
Add the accepted customizer arguments to cloneDeepWith docs.

Crysinyan committed:
Changed to function as documented. (#3596)

JoosAI committed:
Support locales in sorting methods (#3602)

jdalton committed:
Cleanup compareAscending and remove hardcoded locale.

jnfingerle committed:
Update documentation in slice.js (#3631)

Even though #3627 has been closed, negative indexes should be documented.

PxyUp committed:
add uniq Map for generated ID (#3644)

Dean177 committed:
Add string example to isEmpty docs (#3647)

While this is covered in the the description above, there exist examples for arrays and objects and I think this gives 'at a glance' confirmation that the function works as intended with strings.

alirezavalizade committed:
use isObjectLike module instead of (typeof value == 'object' && value !== null) in another modules (#3650)

eran-nussbaum committed:
Implement `defaultToAny` method (#3615)

pahomov committed:
Fix npm link in README.md (#3700)

joyeecheung committed:
internal: use util.types to migrate DEP0103 in Node.js (#3704)

- Use require('util').types instead of using process.binding('util')
  to get the type checking helpers
- Rename nodeUtil to nodeTypes since that is what it is for

Refs: https://github.com/nodejs/node/pull/18415

misogl committed:
Fix inclusion of litteral "+" in rsMisc (#3736)

`rsDigits` contained a "+" character but was also used in negated character class.

Commit:
Update _.union documentation (#3758)

Clarifying example as per https://github.com/lodash/lodash/issues/3757

ecbrodie committed:
Updates version on README header (#3767)

jdalton committed:
Drop version number from readme.

jdalton committed:
Update countBy doc example to be more at a glance friendly. [closes #3771]

ryanhomer committed:
Allow compareMultiple to accept compare functions in addition to order strings (#3764)

twolfson committed:
Added short circuit for typed arrays (#3786)

falsyvalues committed:
Remove trailing space in `slice`

falsyvalues committed:
Remove semicolons from `sampleSize` and `baseClone`

falsyvalues committed:
Tiny style correction in `uniqueId` and `compareMultiple`

jdalton committed:
Avoid skipping set of -0 when an existing +0 exists. [closes #3798]

ycmjason committed:
Create fromEntries.js (#3809)

Rename fromPairs to fromEntries.

renjithspace committed:
Add example of Array slice (#3823)

jdalton committed:
Remove more methods.

jdalton committed:
Remove unused var from tail().

doug-a-brunner committed:
Clarify doc for `range` usage with single arg (#3854)

cooperka committed:
Update `filterObject` docs to reference `pickBy` (#3861)

* Update `filterObject` docs to reference `pickBy`.

dracos committed:
Add extended and supplement combining marks ranges (#3932)

abdullahtariq1171 committed:
add snakeCase example with digit (#3993)

add `snakeCase` example with digit

j3bb9z committed:
Add missing space in uniqWith docblock (#3995)

alireza-mh committed:
add round up description on ceil documentation (#4027)

Signed-off-by: Alireza <alireza.mh@gmail.com>

falsyvalues committed:
Update deps (eslint-plugin-import@2.14.0)

saravyas committed:
converted to fat arrow function (#4059)

alireza-mh committed:
Add lodash-es description for ES6 scripts (#4035)

jdalton committed:
Remove n_ note for unsupported Node.

saravyas committed:
Added arrow function in below files (#4061)

blikblum committed:
Call upperFirst directly, instead of capitalize, in camelCase (#4075) (#4077)

blikblum committed:
Remove support for node v0.10 in arrayLikeKeys (#4079)

blikblum committed:
Remove support for PhantomJS in arrayLikeKeys (#4081)

blikblum committed:
Remove unnecessary null check in arrayEach (#4082)

asilinwei committed:
fix: limit the precision when the precision is negative (#4085)

blikblum committed:
Remove checks for the Map existence (#4091)

blikblum committed:
Replace MapCache by Map in memoize (#4094) (#4095)

blikblum committed:
Restore createAssigner (fixes compilation of merge and mergeWith) (#4101)

blikblum committed:
Remove check for existence of getOwnPropertySymbols (#4102)

blikblum committed:
Re-add keysIn module fixing baseClone and baseMerge (#4099) (#4100)

blikblum committed:
Fix findLast missing dependency (#4103)

blikblum committed:
Fix baseClone missing dependency (#4104)

blikblum committed:
Fix baseOrderBy missing dependency (#4105)

blikblum committed:
Use baseGetTag instead of getTag where possible (#4112)

blikblum committed:
Assume Symbol is always defined (#4111)

blikblum committed:
Remove check for Symbol.toStringTag field mutations in baseGetTag (#4112) (#4113)

blikblum committed:
Replace getTag implementation by the one from baseGetTag (remove workarounds) (#4115)

blikblum committed:
Use native filter instead of custom one where possible (#4116)

jdalton committed:
Add .github/.lock.yml config for lock bot.

moonformeli committed:
Change type check and the way of sending params. (#4145)

moonformeli committed:
Use 3 equal signs (#4153)

Like other if statements, use 3 equal signs

43081j committed:
use extension resolution (#4163)

carbureted committed:
Locally scope key in keysIn (#4173)

moonformeli committed:
Cancel old timer, editing pr #4139 (#4187)

* Cancel old timer, editing pr for $4139

jdalton committed:
Update package.json version.

bnjmnt4n committed:
Add initial test files from lodash v4. (#4172)

* Install test dependencies.

* Add initial test files.

These files were created using a simplistic AST manipulator using `recast` to
preserve formatting. There's bound to be a huge chunk of errors, but this serves
as a good start. QUnit was replaced with Mocha, with ES2015 imports running via
`esm`.

As far as possible, QUnit-specific syntax has been replaced with Mocha's
`describe` and `it`, while the native Node.js `assert` module is used for
assertions. Files in the `test` directory ending in `.test.js` will be treated
as test files.

* Add initial passing files to test run.

stevemoser committed:
Remove --save option as it isn't required anymore (#4199)

* Remove --save option as it isn't required anymore
* Add note about using npm < 5.0.0

gu-xionghong committed:
Fix _.maxBy and _.minBy (#4233)

dwelle committed:
tweak _.clone/_.cloneDeep docs (#4239)

gu-xionghong committed:
Fix findLast missing dependency (#4259)

jdalton committed:
Fix typo in toPlainObject().

Commit:
Minor doc format (#4285)

futurist committed:
Fix stringToPath bug for consecutive [] (#4319)

tkrugg committed:
Fix typo in merge test (#4334)

lvzhenbang committed:
fix reTypedTag error (#4342)

Jack-Works committed:
Add support for globalThis when detecting global object (#4347)

jdalton committed:
Add back accidentally removed semicolons to escape.js.

lirantal committed:
docs(security): add responsible disclosure policy (#4374)

jdalton committed:
Cleanup security.md document.

blikblum committed:
Add missing isTypedArray import to baseClone (#4404)

blikblum committed:
Do not use property shorthand in countBy documentation (#4405)

blikblum committed:
Add missing isObject import to isIterateeCall (#4409)

blikblum committed:
Coerce indexOf fromIndex parameter to integer (#4410)

blikblum committed:
Enable indexOf tests (#4411)

blikblum committed:
Coerce drop and dropRight n param using toInteger (#4412)

* Coerce dropRight n param using toInteger

* Coerce drop n param using toInteger

blikblum committed:
Coerce to integer and set default value for chunk size parameter (#4413)

* Enable chunk module tests

* Coerce to integer and set default value for chunk size param

phapdinh committed:
lint files (#4416)

* lint files

* added back arrow-paren and removed duplicate

blikblum committed:
Implement asciiWords and unicodeWords using String.match instead of binding RegExp.exec (#4417)

* Enable words tests
* Implement asciiWords and unicodeWords using String.match instead of binding RegExp.exec

falsyvalues committed:
Update `eslint` and `eslint-plugin-import`

falsyvalues committed:
Align indentations

blikblum committed:
Ensure after n parameter is coerced from NaN to 0  (#4430)

* Enable after tests

* Ensure after n parameter is coerced from NaN to 0

blikblum committed:
merge: copy function values independent of source index  (#4429)

* merge: copy function values independent of source index

* Enable merge tests

blikblum committed:
Flatten at paths parameter (#4431)

* Enable at tests

* Flatten at paths parameter

blikblum committed:
Enable create tests and adapt it for the new primitive prototype behavior (#4418)

* Enable create tests and adapt it for the new primitive prototype behavior

* Use isObject in create tests instead of manual check

* Store properties in a variable when used more than once in create tests

* Add parens around arrow function params in create tests

blikblum committed:
Ensure escapeRegExp returns an empty string when passing falsy values  (#4432)

* Enable escapeRegExp tests

* Ensure escapeRegExp returns an empty string when passing falsy values

* Add parens around expression in escapeRegExp

blikblum committed:
Enable case methods tests (#4436)

blikblum committed:
Coerce findLastIndex and lastIndexOf fromIndex parameter to integer  (#4440)

* Enable lastIndexOf, findLast and findLastIndex

* Coerce findLastIndex fromIndex parameter to integer

* Coerce lastIndexOf fromIndex parameter to integer

blikblum committed:
Simplify toString (#4441)

* Assume Symbol exists in toString tests

* Use native array map instead of custom one in toString

* Assume Symbol.prototype.toString exists in toString

blikblum committed:
Fix string methods to handle empty values (#4442)

* Enable strings category methods tests

* Ensure escape, pad, padEnd, padStart, trim, trimEnd, trimStart, unescape return an empty string for falsey values

* Coerce value to string using toString in truncate, capitalize and case methods

* Ensure createCaseFirst returns an empty string for falsey values

blikblum committed:
Fix flow and flowRight parameter handling (#4445)

* Enable flow and flowRight tests

* Remove flow and flowRight tests for default value and shortcut fusion

* Use native rest parameters / spread operator in flow and flowRight

* Fix syntax of flow and flowRight examples

blikblum committed:
Prevent ReDos in words (#4450)

Mohit21GoJs committed:
add order by behaviour with array similar to get method (#4453)

* add order by behaviour with array similar to get method

* remove typo for fixing orderby iteratee assignment

* lint fixes

* update test case

* include package-lock from master

* Add identity function for default iteratee

jdalton committed:
Cleanup baseOrderBy.

anatoliisf committed:
perfomance fixes, eslint fixes (#4482)

megawac committed:
Short circuit sortedIndexBy methods for empty arrays (#4496)

prog1dev committed:
Fix vulnerabilities in npm packages (#4490)

falsyvalues committed:
Ensure `orderBy` will accept iteratee path arrays #4438 (#4513)

falsyvalues committed:
Fix parentheses placement broken in 16413cc

camjc committed:
Remove extraneous conditional (#4531)

Since `function flow(funcs)` changed to `function flow(...funcs)` this ternary became unnecessary, since funcs will always have a length now.

RajuPedda committed:
Updated the check of isFunction method (#4555)

blikblum committed:
Enable escape and once tests (#4618)

blikblum committed:
Fix uniqueId prefix coercion (#4619)

gwwar committed:
Handle leading zeros in quote html entity (#4623)

dwesty17 committed:
Update links in CONTRIBUTING.md (#4741)

Keith-CY committed:
fix: fix the jsdoc of SetCache#has (#4739)

The return type should be `boolean` instead of `number`

Keith-CY committed:
fix: fix the params of Hash#delete in in jsdoc (#4737)

Remove the unused param `hash` from the jsdoc of `Hash#delete`

fs-projects committed:
Changing font-style to monospace for '--save' option in README.md (#4681)

--save is a command-line option and it should be monospaced

Co-authored-by: utkarsh-kukreti <utkarshk@dewsolutions.in>

thxiami committed:
Update links in CONTRIBUTING.md & README.md (#4829)

csojinb committed:
Remove outdated contributor tips (#5008)

The `opt-cli` pre-push functionality was removed from lodash just a few days
after it was added (see https://github.com/lodash/lodash/commit/2cd12c38e32c76aca6396df617d79ac752d81935),
but the documentation encouraging contributors to use it still remains.
Remove to avoid confusion for new contributors.

bnjmnt4n committed:
Cleanup package.json and CONTRIBUTING.md.

The master branch has been in flux for a pretty long time, so leaving
the broken npm scripts and messages in CONTRIBUTING.md doesn't make
sense and has been confusing to contributors. This commit removes broken
npm scripts and removes all unnecessary dependencies for now.

bnjmnt4n committed:
Fix ESLint errors and tests.

I've commented out a test for `_.merge` and will re-look it as I
gradually cleanup the codebase.

bnjmnt4n committed:
Enable GitHub Actions CI.

de-ar committed:
Update tests.yml (#5140)

`npm cit` is short for `npm ci` and `npm test`

jacob-lcs committed:
perf: use `===` instead of `==` (#5118)

jdalton committed:
wip: migrate to bun

jdalton committed:
wip: code formatting nits

jdalton committed:
wip: code formatting nits continued

jdalton committed:
wip: code formatting nits continued

tisonkun committed:
ci: make it work (#5728)

Signed-off-by: tison <wander4096@gmail.com>

tisonkun committed:
test: partially fix broken tests (#5733)

* test: fix throttle.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix pickBy.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix isBuffer.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: partially fix attempt.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: partially fix dropRightWhile.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix defer.spec.js and rest.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix invoke.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix isArray.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: partially fix iteration-methods.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix xor-methods.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix property.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix ary.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix omit-methods.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix debounce-and-throttle.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix unzip-and-zip.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix toPairs-methods.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix exit-early.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: temporarily comment out takeWhile and dropWhile tests

Signed-off-by: tison <wander4096@gmail.com>

* test: partially fix union*.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix startsWith-and-endsWith.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix isNil.spec.js

Signed-off-by: tison <wander4096@gmail.com>

* test: fix some of syntax errors

Signed-off-by: tison <wander4096@gmail.com>

---------

Signed-off-by: tison <wander4096@gmail.com>

jdalton committed:
wip: unit test fixes continued

tisonkun committed:
test: fix cannot find root (#5741)

Signed-off-by: tison <wander4096@gmail.com>

jinbekim committed:
test: add some cases (#5789)

ygj6 committed:
fix: inRange test (#5821)

Commit:
Added default value field to toNumber function

Trott commented:
Needs tests and docs.

Doesn't seem like a great API function signature. If I look at code and see `_.toNumber(something)`, that's clear. If I see `_.toNumber(something, somethingElse)`, I'm not sure what is going on. I'm not sure about the performance implications, but it is probably better to use an options object in this case: `_.toNumber(something, {defaultValue: 0})`

kapilkumar-github committed:
feat: added options to tonumber method

1) Added options as second param for toNumber
2) defaultValue will be returned in case if NaN occurs
3) Updated example and docs for toNumber

kapilkumar-github committed:
test: Added test cases for toNumber method

kapilkumar9395 commented:
@Trott thanks for the input. I have made the necessary changes

Trott commented:
> @Trott thanks for the input. I have made the necessary changes

I'm not a lodash maintainer or anything, so bear that in mind, but I would imagine that the tests should be added to `test/number-coercion-methods.spec.js` rather than creating a separate test file just for additional `toNumber()` tests. That also raises the question of whether symmetric changes should be made to the API of `toFinite()`, `toInteger()`, and  `toSafeInteger()`. But those never return `NaN` so I'm not sure it makes sense.

That in turn raises the issue of whether this API adjustment makes sense. It seems like it would typically be used to return `0` instead of `NaN`, but that's already what `toFinite()` does, and `toFinite()` has the added bonus of returning usable values for `Infinity` and `-Infinity`.
